src/components/calcite-rating/calcite-rating.tsx

```diff
@@ -0,0 +1,385 @@
+    const iconsToRender = [];
+    for (let i = 0; i < 5; i++) {
+      iconsToRender.push(
+        <calcite-icon
+          data-value={i + 1}
+          onClick={(e) => this.determineRatingPosition(e)}
+          onKeyDown={(e) => this.handleKeyDown(e)}
+          onMouseEnter={(e) => this.showSelectedIconOnHover(e)}
+          onTouchStart={(e) => this.showSelectedIconOnHover(e)}
+          onFocus={(e) => this.showSelectedIconOnHover(e)}
+          tabindex={!this.readOnly && !this.disabled ? 0 : null}
+          icon={this.iconType}
+          scale={this.scale}
+        />
+      );
+    }
+    return (
+      <Host dir={this.dir}>
+        <div
+          class="calcite-rating-item-container"
+          onBlur={() => this.resetHoverState()}
+          onMouseLeave={() => this.resetHoverState()}
+          onTouchEnd={() => this.resetHoverState()}
+        >
+          {iconsToRender}
+          {this.average && !this.value ? partialRatingStarContainer : null}
+        </div>
+        {this.count || this.average ? (
+          <calcite-chip
+            dir={this.dir}
+            theme={this.theme}
+            scale={this.scale}
+            value={this.count?.toString()}
+          >
+            {this.average ? (
+              <span class="calcite-rating-average">{this.average.toString()}</span>
+            ) : null}
+            {this.count ? (
+              <span class="calcite-rating-count">({this.count?.toString()})</span>
+            ) : null}
+          </calcite-chip>
+        ) : null}
+      </Host>
+    );
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private handleKeyDown(e) {
+    if (!this.readOnly) {
+      const itemToFocus = e.target;
+      const isFirstItem = this.itemIndex(itemToFocus) === 0;
+      const isLastItem = this.itemIndex(itemToFocus) === this.ratingItems.length - 1;
+      switch (getKey(e.key)) {
+        case " ":
+        case "Enter":
+          e.preventDefault();
+          this.determineRatingPosition(e);
+          break;
+        case "ArrowRight":
+          if (isLastItem) this.focusFirstItem();
+          else this.focusNextItem(itemToFocus);
+          break;
+        case "ArrowLeft":
+          if (isFirstItem) this.focusLastItem();
+          else this.focusPrevItem(itemToFocus);
+          break;
+        case "Home":
+          this.focusFirstItem();
+          e.preventDefault();
+          break;
+        case "End":
+          this.focusLastItem();
+          e.preventDefault();
+          break;
+        case "Tab":
+          if (isLastItem || (isFirstItem && e.shiftKey)) this.resetHoverState();
+          break;
+      }
+    }
+  }
+
+  private determineRatingPosition(e) {
+    if (!this.readOnly) {
+      // handle edge and ie not interpreting mapped stars
+      const isEdgeOrIE =
```

Comment:
nitpick: this kinda makes me squeamish. Is there a way to use feature detection instead of browser detection? I'm worried this could be problematic in the future.

Reply:
Maybe instead of storing the value on the icon, there could be an interface for the array of icons that handles that?

```
ratingItems: {icon: HTMLCalciteIconElement, value: number,...}
```

Reply:
I guess the other solution would be to create a `calcite-rating-item` that used `calcite-icon` internally and added props for value, hovered, average, selected?

Reply:
Or a functional component would maybe be best since it wouldn't be public.

Reply:
Re: `isEdgeOrIE` - Luckily I can strip all that out - Legacy Edge was struggling to interpret the type of element rendered through the rendered map of stars, that's not there anymore since IE can't handle it at all so I can remove these references.

Reply:
Re: value - you mean instead of `data-value`? Yeah, that makes sense to me

Reply:
Yeah, I was just thinking of storing them either in a typed object or functional component instead of using data attributes.